src/apps/filters/geoToH3.c

```diff
@@ -34,40 +34,80 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include "coordijk.h"
 #include "h3Index.h"
 #include "utility.h"
 
 int main(int argc, char* argv[]) {
-    // get the command line argument resolution
-    if (argc != 2) {
-        fprintf(stderr, "usage: %s resolution\n", argv[0]);
-        exit(1);
-    }
+    int res = 0;
+    double lat = 0;
+    double lon = 0;
 
-    int res;
-    if (!sscanf(argv[1], "%d", &res)) error("parsing resolution");
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText = "Resolution, 0-15 inclusive."};
+    Arg latArg = {.names = {"-lat", "--latitude"},
```

Comment:
Nit: I'd have thought `--lat` and `--lon` would be the short forms - I've generally seen single-dash only for single-char args (partly so that if we wanted we could stack like `-xvf`).

One one-char option would be to use `-y` for lat and `-x` for long.

Reply:
Do you have a preference between adding `--lat`/`--lon` or `-y`/`-x`? I am leaning towards `--lat`/`--lon` since it does not have the ambiguity about which comes first.

Reply:
To me -y / -x sounds like either planar coordinates or two components of 3D ECEF coordinates. Both might be potential arguments to some future program (especially given that H3 can now output planar IJK grids).